master/internal/api_trials.go

```diff
@@ -74,12 +74,6 @@ func (a *apiServer) TrialLogs(
 		return status.Error(codes.InvalidArgument, fmt.Sprintf("unsupported filter: %s", err))
 	}
 
-	total, err := a.m.trialLogBackend.TrialLogsCount(int(req.TrialId), filters)
-	if err != nil {
-		return fmt.Errorf("failed to get trial count from backend: %w", err)
-	}
-	offset, limit := api.EffectiveOffsetNLimit(int(req.Offset), int(req.Limit), total)
```

Comment:
We don't calculate effective limit any, which was just min(log_count, limit), but this is fine; if limit is less, this is the same behavior, if it is more, the `BatchStreamProcessor` will terminate when it gets a nil or len(batch) == 0 batch.